lib/grunt/task.js

```diff
@@ -370,7 +370,23 @@ task.loadTasks = function(tasksdir) {
 task.loadNpmTasks = function(name) {
   loadTasksMessage('"' + name + '" local Npm module');
   var root = path.resolve('node_modules');
-  var pkgfile = path.join(root, name, 'package.json');
+  var pkgpath = path.join(root, name);
+  var pkgfile = path.join(pkgpath, 'package.json');
+  // If package does not exist where grunt expects it to be,
+  // try to find it using Node's package path resolution mechanism
+  if (!grunt.file.exists(pkgpath)) {
+    var nameParts = name.split('/');
```

Comment:
@redonculus , I tested my PR on windows and it works fine. There is a need to use `path.sep` only if someone is using syntax like `require(".\\my\\foo\\index.js")`. While this may work on windows, it fails on other platform. In opposite, using `require("./my/foo/index.js")` works on all platforms (including windows). So, practically, I don't see any reason someone would use windows specific paths, if there is a way to write cross-platform code with less typing.

Reply:
Ok that’s fine then. If it works on windows then I’m good.